plugins/inputs/logfile/logfile_test.go

```diff
@@ -358,100 +358,128 @@ func TestLogsFileRemove(t *testing.T) {
 	tt.Stop()
 }
 
-//When another file is created for the same file config and the file config has auto_removal as true, the old files will stop at EOF and removed afterwards
-func TestLogsFileAutoRemoval(t *testing.T) {
-	var wg sync.WaitGroup
+// getLogSrc returns a LogSrc from the given LogFile, and the channel for output.
+// Verifies 1 and only 1 LogSrc is discovered.
+func getLogSrc(t *testing.T, logFile *LogFile) (*logs.LogSrc, chan logs.LogEvent) {
+	logSources := logFile.FindLogSrc()
+	require.Equal(t, 1, len(logSources))
+	logSource := logSources[0]
+	evts := make(chan logs.LogEvent)
+	logSource.SetOutput(func(e logs.LogEvent) {
+		if e != nil {
+			evts <- e
+		}
+	})
+	return &logSource, evts
+}
 
+func writeSomeLines(t *testing.T, file *os.File, numLines int, msg string) {
+	for i := 0; i < numLines; i++ {
+		//time.Sleep(1 * time.Millisecond)
+		//fmt.Printf("write i %v\n", i)
+		_, err := file.WriteString(msg + "\n")
+		require.NoError(t, err)
+	}
+}
+
+// TestLogsFileAutoRemoval verifies when a new file matching the configured
+// FilePath is discovered, the old file will be automatically deleted after
+// being read to the end-of-file.
+func TestLogsFileAutoRemoval(t *testing.T) {
+	// Override global in tailersrc.go.
 	multilineWaitPeriod = 10 * time.Millisecond
-	logEntryString := "anything"
+
 	filePrefix := "file_auto_removal"
 	tmpfile1, err := createTempFile("", filePrefix)
+	fmt.Printf("Created 1st temp file, %s\n", tmpfile1.Name())
 	require.NoError(t, err)
 
-	_, err = tmpfile1.WriteString(logEntryString + "\n")
-	require.NoError(t, err)
-	tmpfile1.Sync()
-	tmpfile1.Close()
-
-	tt := NewLogFile()
-	tt.Log = TestLogger{t}
-	tt.FileConfig = []FileConfig{{
+	logFile := NewLogFile()
+	defer logFile.Stop()
+	logFile.Log = TestLogger{t}
+	logFile.FileConfig = []FileConfig{{
 		FilePath:      filepath.Join(filepath.Dir(tmpfile1.Name()), filePrefix+"*"),
 		FromBeginning: true,
 		AutoRemoval:   true,
 	}}
-	tt.FileConfig[0].init()
-	tt.started = true
+	logFile.FileConfig[0].init()
+	logFile.started = true
```

Comment:
not super important, but looks like this set up (minus the defer `Stop()`) _could_ be extracted into a helper, but you'd only save like 5 lines of code in this test.